core/src/main/java/org/frankframework/stream/Message.java

```diff
@@ -294,29 +293,29 @@ private void preserve(boolean deepPreserve) throws IOException {
 	}
 
 	private void preserveToMemory(boolean deepPreserve) throws IOException {
-		if (request instanceof SerializableFileReference) {
+		if (messageCleanupAction.request instanceof SerializableFileReference) {
 			// Should not happen but just in case.
 			return;
 		}
-		if (request instanceof Reader reader) {
+		if (messageCleanupAction.request instanceof Reader reader) {
 			LOG.debug("preserving Reader {} as String", this::getObjectId);
-			request = StreamUtil.readerToString(reader, null);
+			messageCleanupAction.request = StreamUtil.readerToString(reader, null);
```

Comment:
Ik snap dat dit niet helemaal jouw aanpassing is, maar: ik zou verwachten dat mutaties op private members van een class (messageCleanupAction) niet direct gemuteerd kunnen worden, maar alleen via een setX of addX methode.